core/src/main/java/nu/yona/server/analysis/entities/IntervalActivity.java

```diff
@@ -12,16 +17,44 @@
 @MappedSuperclass
 public abstract class IntervalActivity extends EntityWithID
 {
+	public static final int SPREAD_COUNT = 96;
+
 	@ManyToOne
 	private Goal goal;
 
+	/*
+	 * The date. Stored for easier querying (if the time zone of the user changes, we cannot query for equal start time).
+	 */
+	private LocalDate date;
+
+	/*
+	 * The start date and time with saved time zone.
+	 */
 	private ZonedDateTime startTime;
 
-	protected IntervalActivity(UUID id, Goal goal, ZonedDateTime startTime)
+	@ElementCollection
+	private List<Integer> spread;
```

Comment:
Maybe an alternative is to use a `String` here and use a `@Transient` array that is joined in this string (e.g. comma-separated).